nixos/modules/virtualisation/virtualbox-host.nix

```diff
@@ -28,7 +29,16 @@ in
       '';
     };
 
-    enableExtensionPack = mkEnableOption "VirtualBox extension pack";
+    enableExtensionPack = mkEnableOption "VirtualBox extension pack" // {
+      description = ''
+        Whether to install the Oracle Extension Pack for VirtualBox.
+
+        <important><para>
+          You must set <literal>nixpkgs.config.allowUnfree = true</literal> in
+          order to use this.  This requires you accept the VirtualBox PUEL.
+        </para></important>
```

Comment:
How is the license being accepted? It seems like it's possible to  set `config.allowUnfree = true` without having read the license. It's a blanked agreement to all unfree licenses. Usually EULA require an explicit approval (which is questionable, but that's what is happening).

Reply:
One idea would be that the user has to provide the sha256 of the license as a "proof" that he read and accepted it

Reply:
Hi @zimbatm, thanks for taking a look at this PR.

> How is the license being accepted?

Originally (the current state of nixpkgs), if you enable `virtualization.virtualbox.host.enableExtensionPack`, `nix-build` will fail and you are presented with a message telling you to read the PUEL and asking you to download the download the Oracle Extension Pack directly by using `nix-prefetch-url`.

This PR changes it so that the license is accepted and the Extension Pack will be downloaded during `nix-build` if you set both `config.allowUnfree = true` and `virtualization.virtualbox.host.enableExtensionPack = true`.

> One idea would be that the user has to provide the sha256 of the license as a "proof" that he read and accepted it

As I said in the original issue, I don't know the legal ramifications of setting it up to work one way vs another way.

However, I'm not sure I understand the reasoning for handling the PUEL as a somewhat special case, and requiring the end user to do something like provide a `sha256` of the license.

I took a look at another random non-free license, and it seems to have similar wording to the PUEL.  Here is the license for the Elastic License, which is used in the non-free version of kibana, logstash, etc:

https://github.com/elastic/elasticsearch/blob/master/licenses/ELASTIC-LICENSE.txt

Here is the PUEL for comparison:

https://www.virtualbox.org/wiki/VirtualBox_PUEL

Here is the page that allows you download the Extension Pack and links to the above license:

https://github.com/elastic/elasticsearch/blob/master/licenses/ELASTIC-LICENSE.txt

----------------------------------------------------------------------------------------

With that said, if everyone agrees that the PUEL is special in this regard, and we should do something like require the user to provide a sha256 of the license text, then I would be happy to change this PR to do that!

My main goal is to get rid of the manual steps in installing Virtualbox with the Extension Pack.

Reply:
Okay so this is complementary. `pkgs/stdenv/generic/check-meta.nix` there is a `allowUnfreePredicate` that would allow to build a list of authorized licenses.

I think the issue is around the notion of explicit consent, a user should not be able to use software on which he hasn't agreed on the license. `allowUnfree = true` is quite a low bar to say "Yes I have read and understand all the unfree license and I agree with all of them".

But like you say the mechanism is like that currently, it's independent and shouldn't hold back this PR.